### ISSUE TITLE
Significantly different loss between legacy and ADF options using CB with cover

### ISSUE THREAD

**Issue by maxpagels** [VowpalWabbit/vowpal_wabbit#3704]
#### Describe the bug
Since VW 8.11., cb-type data is automatically converted to ADF under the hood. Legacy behaviour can be enabled with `--cb_force_legacy`. However, with `--cover` and the default cb_type, the reported loss is significantly different between legacy / non-legacy. In addition, `--indexing` gives a segfault in certain situations.

#### To Reproduce
1. Download [repro.txt](https://github.com/VowpalWabbit/vowpal_wabbit/files/8031954/repro.txt)
2. Run `python -m vowpalwabbit --cb_explore 2 --cover 4 -d repro.txt` and observe `average loss = 0.001078`
3. Run ` python -m vowpalwabbit --cb_explore 2 --cover 4 -d repro.txt --indexing 1 --cb_force_legacy`  and observe `average loss = 0.003236`
4. Bonus: run `python -m vowpalwabbit --cb_explore 2 --cover 4 -d repro.txt --indexing 0 --cb_force_legacy` and observe label warning (expected) followed by `Segmentation fault: 11`

#### Expected behavior
Approximately similar loss

#### Observed Behavior
Significantly different loss; is this due to `[warning] currently, mtr is only used for the first policy in cover, other policies use dr` or a proper bug?

#### Environment
VW version: Python package, 9.0.0
Mac OS 11.5.2 Big Sur

**Comment by jackgerrits** [comment]
That route was not fruitful

**Comment by bassmang** [comment]
Hi @maxpagels,

In reference to the indexing issue, I've put out a PR to support 0-indexed examples with cb_explore. Note that while your example above will not segfault anymore, it will skip over and output warnings for examples with the label 2. When using `--cb_explore 2` and `--indexing 0` only labels `0` and `1` can be processed properly.

**Comment by maxpagels** [comment]
@bassmang great, thanks for the explanation. Feel free to close.

**Comment by bassmang** [comment]
Closed with addition of printing cb_type and keeping default as `dr` for legacy and `mtr` for cb_adf.
